x-pack/legacy/plugins/lens/public/indexpattern_plugin/indexpattern_suggestions.tsx

```diff
@@ -298,17 +298,73 @@ function createNewLayerWithMetricAggregation(
 export function getDatasourceSuggestionsFromCurrentState(
   state: IndexPatternPrivateState
 ): Array<DatasourceSuggestion<IndexPatternPrivateState>> {
-  const layers = Object.entries(state.layers);
+  const layers = Object.entries(state.layers || {});
 
-  return layers
-    .map(([layerId, layer], index) => {
+  return _.flatten(
+    layers.map(([layerId, layer], index) => {
       if (layer.columnOrder.length === 0) {
-        return;
+        return [];
       }
 
-      return buildSuggestion({ state, layerId, isMultiRow: true, datasourceSuggestionId: index });
+      const onlyMetric = layer.columnOrder.every(columnId => !layer.columns[columnId].isBucketed);
+      const onlyBucket = layer.columnOrder.every(columnId => layer.columns[columnId].isBucketed);
+      if (onlyMetric || onlyBucket) {
+        // intermediary chart, don't try to suggest reduced versions
+        return [
+          buildSuggestion({ state, layerId, isMultiRow: false, datasourceSuggestionId: index }),
+        ];
+      }
+
+      return createSimplifiedTableSuggestions(state, layerId);
+    })
+  ).map(
+    (suggestion, index): DatasourceSuggestion<IndexPatternPrivateState> => ({
+      ...suggestion,
+      table: { ...suggestion.table, datasourceSuggestionId: index },
+    })
+  );
+}
+
+function createSimplifiedTableSuggestions(state: IndexPatternPrivateState, layerId: string) {
+  const layer = state.layers[layerId];
+
+  const availableBucketedColumns = layer.columnOrder.filter(
+    columnId => layer.columns[columnId].isBucketed
+  );
+  const availableMetricColumns = layer.columnOrder.filter(
+    columnId => !layer.columns[columnId].isBucketed
+  );
+
+  return _.flatten(
+    availableBucketedColumns.map((_col, index) => {
+      const bucketedColumns = availableBucketedColumns.slice(0, index + 1);
+      const allMetricsSuggestion = buildLayerByColumnOrder(layer, [
+        ...bucketedColumns,
+        ...availableMetricColumns,
+      ]);
+
+      if (availableMetricColumns.length > 1) {
+        return [
+          allMetricsSuggestion,
+          buildLayerByColumnOrder(layer, [...bucketedColumns, availableMetricColumns[0]]),
+        ];
+      } else {
+        return [allMetricsSuggestion];
```

Comment:
Same comment about not needing to return an array, but again, not super important.